frontend/components/Filter/OrganiserFilterDialog.tsx

```diff
@@ -0,0 +1,415 @@
+  };
+  const toggleShowSearch = () => {
+    setShowSearch(!showSearch);
+  };
+  const toggleShowEventStatus = () => {
+    setShowEventStatus(!showEventStatus);
+  };
+  const toggleShowEventType = () => {
+    setShowEventType(!showEventType);
+  };
+  const toggleShowPriceRange = () => {
+    setShowPriceRange(!showPriceRange);
+  };
+  const toggleShowDateRange = () => {
+    setShowDateRange(!showDateRange);
+  };
+
+  const [sortByKey, setSortByKey] = useState(0);
+
+  const updateSortByKey = () => {
+    setSortByKey((prevKey) => prevKey + 1);
+  };
+
+  const handleDateRangeChange = (dateRange: any) => {
+    if (dateRange.startDate && dateRange.endDate) {
+      let timestampDateRange = {
+        startDate: dateRange.startDate,
+        endDate: dateRange.endDate,
+      };
+
+      setDateRange(timestampDateRange);
+    } else {
+      let timestampDateRange = {
+        startDate: DEFAULT_START_DATE,
+        endDate: DEFAULT_END_DATE,
+      };
+
+      setDateRange(timestampDateRange);
+    }
+  };
+
+  const [datepickerKey, setDatepickerKey] = useState(0);
+
+  const updateDatepickerKey = () => {
+    setDatepickerKey((prevKey) => prevKey + 1);
+  };
+
+  const DatepickerComponent = useMemo(
+    () => (
+      <Datepicker
+        key={datepickerKey}
+        value={dateRange}
+        separator="to"
+        displayFormat={"DD/MM/YYYY"}
+        onChange={handleDateRangeChange}
+        inputClassName="w-full p-2 border rounded-xl focus:outline-none focus:ring focus:border-blue-300"
+      />
+    ),
+    [datepickerKey, dateRange]
+  );
+
+  const toggleStatusCheckboxValue = (value: string) => {
+    if (eventStatusValue.includes(value)) {
+      setEventStatusValue(eventStatusValue.replace(value, ""));
+    } else {
+      setEventStatusValue(eventStatusValue + value);
+      if (eventStatusValue.includes("true") && eventStatusValue.includes("false")) {
+        setEventStatusValue("");
+      }
+    }
+  };
+
+  const toggleTypeCheckboxValue = (value: string) => {
+    if (eventTypeValue.includes(value)) {
+      setEventTypeValue(eventTypeValue.replace(value, ""));
+    } else {
+      setEventTypeValue(eventTypeValue + value);
+    }
+    if (eventTypeValue.includes("true") && eventTypeValue.includes("false")) {
+      setEventStatusValue("");
+    }
+  };
+
+  function handleClearAll() {
+    setSortByCategoryValue(DEFAULT_SORT_BY_CATEGORY);
+    updateSortByKey();
+    setSearchValue(DEFAULT_SEARCH);
+    setEventStatusValue(DEFAULT_EVENT_STATUS);
+    setEventTypeValue(DEFAULT_EVENT_TYPE);
+    setMinPriceValue(DEFAULT_MIN_PRICE);
+    setMaxPriceValue(DEFAULT_MAX_PRICE);
+    setDateRange({
+      startDate: DEFAULT_START_DATE,
+      endDate: DEFAULT_END_DATE,
+    });
+    updateDatepickerKey();
+    setEventDataList([...allEventsDataList]);
+  }
+
+  return (
+    <div className="w-[360px] mr-2 2xl:mr-6 max-h-screen">
+      <div className="p-4 bg-gray-100 border border-gray-300 rounded-lg">
```

Comment:
Ideally, we want to be standardising the colors used throughout organiser platform (and realistically the whole sight). Would be great if you could add these colors to the tailwind.config.js file under colors: organiser.

Reply:
picked some rando colours, was waiting for a particular colour to be chosen to implement

Reply:
pre mvp but after organiser platform is finished